src/client/app/containers/app.ts

```diff
@@ -190,22 +190,24 @@ export class AppComponent {
                     this._store.dispatch(new GitHub.UpdateGistAction(this.snippet));
                 }
                 else if (isPublic) {
-                    this._effects.alert(this.strings.sharePublicSnippetConfirm, `${this.strings.share} ${this.snippet.name}`, this.strings.share, this.strings.cancelButtonLabel)
-                        .then((result: string) => {
-                            if (result !== this.strings.cancelButtonLabel) {
-                                this._store.dispatch(new GitHub.SharePublicGistAction(this.snippet));
-                            }
+                    if (this.snippet.gist) {
+                        let result = await this._effects.alert(this.strings.sharePublicSnippetConfirm, `${this.strings.share} ${this.snippet.name}`, this.strings.share, this.strings.cancelButtonLabel);
+                        if (result === this.strings.cancelButtonLabel) {
```

Comment:
But it's shared between both statements.  So I think you can just have a `let confirmationAlertIfAny` defined above, set that variable with the `await this._effects.alert(...)`, and do the if check in one common spot below.